### PR TITLE
Add table `azure_api_management_backend`. Closes #644

### LINKED ISSUE #644: Add table azure_api_management_backend
**References**
https://registry.terraform.io/providers/hashicorp/azurerm/latest/docs/resources/api_management_backend

List API: https://github.com/Azure/azure-sdk-for-go/blob/v58.0.0/services/apimanagement/mgmt/2020-12-01/apimanagement/backend.go#L417
Get API: https://pkg.go.dev/github.com/Azure/azure-sdk-for-go@v58.0.0+incompatible/services/apimanagement/mgmt/2020-12-01/apimanagement#BackendClient.Get

Note: This is required for new query development.

### PR DESCRIPTION
**Pull request by madhushreeray30**
# Integration test logs
<details>
  <summary>Logs</summary>
  
```
Add passing integration test logs here
```
</details>

# Example query results
<details>
  <summary>Results</summary>
  
```sql
select * from azure_api_management_backend
[
 {
  "_ctx": {
   "connection_name": "azure"
  },
  "akas": [
   "azure:///subscriptions/d46d7416-2222-3333-bbb5-111111111111/resourceGroups/DefaultResourceGroup-WEU/providers/Microsoft.ApiManagement/service/mrtest/backends/mrtestbackend",
   "azure:///subscriptions/d46d7416-2222-3333-bbb5-111111111111/resourcegroups/defaultresourcegroup-weu/providers/microsoft.apimanagement/service/mrtest/backends/mrtestbackend"
  ],
  "backend_id": "mrtestbackend",
  "cloud_environment": "AzurePublicCloud",
  "credentials": {
   "authorization": {
    "parameter": "username:password",
    "scheme": "Basic"
   },
   "header": {},
   "query": {}
  },
  "description": "This is the test backend",
  "id": "/subscriptions/d46d7416-2222-3333-bbb5-111111111111/resourceGroups/DefaultResourceGroup-WEU/providers/Microsoft.ApiManagement/service/mrtest/backends/mrtestbackend",
  "name": "mrtestbackend",
  "properties": null,
  "protocol": "http",
  "proxy": null,
  "resource_group": "defaultresourcegroup-weu",
  "resource_id": null,
  "service_name": "mrtest",
  "subscription_id": "d46d7416-2222-3333-bbb5-111111111111",
  "title": "mrtestbackend",
  "tls": {
   "validateCertificateChain": true,
   "validateCertificateName": true
  },
  "type": "Microsoft.ApiManagement/service/backends",
  "url": "https://api.sampleproductservice.com/v1/products"
 }
]
```

```sql
select * from azure_api_management_backend where backend_id='mrtestbackend' and service_name = 'mrtest' and resource_group = 'defaultresourcegroup-weu'
[
 {
  "_ctx": {
   "connection_name": "azure"
  },
  "akas": [
   "azure:///subscriptions/d46d7416-2222-3333-bbb5-111111111111/resourceGroups/defaultresourcegroup-weu/providers/Microsoft.ApiManagement/service/mrtest/backends/mrtestbackend",
   "azure:///subscriptions/d46d7416-2222-3333-bbb5-111111111111/resourcegroups/defaultresourcegroup-weu/providers/microsoft.apimanagement/service/mrtest/backends/mrtestbackend"
  ],
  "backend_id": "mrtestbackend",
  "cloud_environment": "AzurePublicCloud",
  "credentials": {
   "authorization": {
    "parameter": "username:password",
    "scheme": "Basic"
   },
   "header": {},
   "query": {}
  },
  "description": "This is the test backend",
  "id": "/subscriptions/d46d7416-2222-3333-bbb5-111111111111/resourceGroups/defaultresourcegroup-weu/providers/Microsoft.ApiManagement/service/mrtest/backends/mrtestbackend",
  "name": "mrtestbackend",
  "properties": null,
  "protocol": "http",
  "proxy": null,
  "resource_group": "defaultresourcegroup-weu",
  "resource_id": null,
  "service_name": "mrtest",
  "subscription_id": "d46d7416-2222-3333-bbb5-111111111111",
  "title": "mrtestbackend",
  "tls": {
   "validateCertificateChain": true,
   "validateCertificateName": true
  },
  "type": "Microsoft.ApiManagement/service/backends",
  "url": "https://api.sampleproductservice.com/v1/products"
 }
]

```
</details>

### CODE REVIEW

File: azure/table_azure_api_management_backend.go [hunk 1]
```diff
@@ -0,0 +1,225 @@
...
+		Description: "Azure API Management Backend",
+		Get: &plugin.GetConfig{
+			KeyColumns: plugin.AllColumns([]string{"backend_id", "resource_group", "service_name"}),
+			Hydrate:    getAPIManagementBackend,
+			IgnoreConfig: &plugin.IgnoreConfig{
+				ShouldIgnoreErrorFunc: isNotFoundError([]string{"ResourceNotFound", "ResourceGroupNotFound"}),
+			},
+		},
+		List: &plugin.ListConfig{
+			ParentHydrate: listAPIManagements,
```
Comment (ParthaI) [inline azure/table_azure_api_management_backend.go:28]: @madhushreeray30, Can you please add `service_name`, `resource_group` and other filter parameter [supported by API](https://learn.microsoft.com/en-us/rest/api/apimanagement/backend/list-by-service?view=rest-apimanagement-2022-08-01&tabs=HTTP) for optimizing the API calls?

File: azure/table_azure_api_management_backend.go [hunk 2]
```diff
@@ -0,0 +1,225 @@
...
+			},
+			{
+				Name:        "id",
+				Description: "Contains ID to identify an API management backend uniquely.",
+				Type:        proto.ColumnType_STRING,
+				Transform:   transform.FromGo(),
+			},
+			{
+				Name:        "url",
+				Description: "Runtime Url of the Backend.",
```
Comment (ParthaI) [inline azure/table_azure_api_management_backend.go:45]: Could you please ensure consistent usage of the term `API management backend` across all column descriptions?

File: azure/table_azure_api_management_backend.go [hunk 3]
```diff
@@ -0,0 +1,225 @@
...
+			},
+			{
+				Name:        "description",
+				Description: "Backend Description.",
+				Type:        proto.ColumnType_STRING,
+				Transform:   transform.FromField("BackendContractProperties.Description"),
+			},
+			{
+				Name:        "resource_id",
+				Description: "Backend Description.",
```
Comment (ParthaI) [inline azure/table_azure_api_management_backend.go:68]: Is this description correct?

File: azure/table_azure_api_management_backend.go [hunk 4]
```diff
@@ -0,0 +1,225 @@
...
+)
+
+//// TABLE DEFINITION ////
+
+func tableAzureAPIManagementBackend(_ context.Context) *plugin.Table {
+	return &plugin.Table{
+		Name:        "azure_api_management_backend",
+		Description: "Azure API Management Backend",
+		Get: &plugin.GetConfig{
+			KeyColumns: plugin.AllColumns([]string{"backend_id", "resource_group", "service_name"}),
```
Comment (ParthaI) [inline azure/table_azure_api_management_backend.go:21]: Can we consider using just one required column, `id`, as the mandatory parameter, and then parse this `id` to extract `backend_id`, `resource_group`, and `service_name`?

File: azure/table_azure_api_management_backend.go [hunk 5]
```diff
@@ -0,0 +1,225 @@
...
+		}
+	}
+
+	return nil, err
+}
+
+//// HYDRATE FUNCTIONS
+
+func getAPIManagementBackend(ctx context.Context, d *plugin.QueryData, h *plugin.HydrateData) (interface{}, error) {
+	plugin.Logger(ctx).Debug("getAPIManagementBackend")
```
Comment (ParthaI) [inline azure/table_azure_api_management_backend.go:196]:
```suggestion
```

File: azure/table_azure_api_management_backend.go [hunk 6]
```diff
@@ -0,0 +1,225 @@
...
+		return nil, err
+	}
+	subscriptionID := session.SubscriptionID
+
+	apiManagementBackendClient := apimanagement.NewBackendClientWithBaseURI(session.ResourceManagerEndpoint, subscriptionID)
+	apiManagementBackendClient.Authorizer = session.Authorizer
+
+	op, err := apiManagementBackendClient.Get(ctx, resourceGroup, serviceName, backendID)
+	if err != nil {
+		plugin.Logger(ctx).Error("getAPIManagementBackend", "get", err)
```
Comment (ParthaI) [inline azure/table_azure_api_management_backend.go:220]: Please update the log statements per the Steampipe standard(`"<table name>.<function name>", "client/api error", err`)?

**Review by ParthaI — COMMENTED** [review]
@madhushreeray30, Please take a look at the review comments. Thanks!

File: azure/table_azure_api_management_backend.go (hunk 5 above)
Comment (ParthaI) [reply azure/table_azure_api_management_backend.go:196]: Removed the unnecessary log statements.

**Review by khushboo9024 — APPROVED** [review]

**Pull request merged**